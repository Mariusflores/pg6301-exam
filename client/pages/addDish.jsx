import {useState} from "react";
import {useNavigate} from "react-router-dom";


export function AddDish(){

    const navigate = useNavigate();

    const [title, setTitle] = useState("");
    const [description, setDescription] = useState("")

    async function handleSubmit(e) {
        e.preventDefault();

        await fetch("/api/dishes", {
            method:"POST",
            headers: {
                "content-type": "application/json"
            },
            body:JSON.stringify({title, description})
        })
        navigate("/menu")
    }

    return <div>
        <form onSubmit={handleSubmit}>
            <h1>Add new dish</h1>

            <div className={"input"}>
                <label>Dish <input type="text" onChange={e => setTitle(e.target.value)}/></label>
                <label> Description <input type="text" onChange={e => setDescription(e.target.value)}/></label>
            </div>
            <button>Submit</button>
        </form>
    </div>
}