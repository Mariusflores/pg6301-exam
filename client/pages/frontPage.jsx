import * as React from "react";

export function FrontPage() {
    return <div>
        <h1>Welcome to Kristiania Catering!</h1>


    </div>;
}