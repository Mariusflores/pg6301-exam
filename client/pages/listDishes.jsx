import {useEffect, useState} from "react";
import * as React from "react"

function useLoading(loadingFunction){
    const [loading, setLoading] = useState(true);
    const [error, setError] = useState();
    const[data, setData] = useState();

    async function load(){

        try{
            setLoading(true);
            setData(await loadingFunction())
        }catch (error){
            setError(error);
        }finally {
            setLoading(false);
        }
    }

    useEffect( () => {
        load();
    }, [])

    return{loading, error, data, reload:load}
}

async function fetchJSON(url){

    const res = await fetch(url)

    if (!res.ok){
        throw new Error(`Failed to load ${res.status} : ${res.statusText}`)
    }

    return await res.json();
}

export function ListDishes(){

    const {loading, error, data} = useLoading(
        async() => fetchJSON("/api/dishes")
    )

    if (loading){
        return <div>Loading..</div>
    }
    if (error){
        return <div>
            <h1>Error</h1>
            <div>{error.toString()}</div>
        </div>

    }
    return <div>

        <h1>Menu</h1>

        <ul>
            {data.map(dish => (
                <li key  ={dish.id}>
                    <h3>{dish.title}</h3>
                    <p>{dish.description}</p>
                </li>
            ))}
        </ul>

    </div>



}