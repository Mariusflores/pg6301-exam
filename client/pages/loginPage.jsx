
export function LoginPage(){

    return <div>
        <form action={"/login"} method={"post"}>
            <div>username:<input type="text" name={"username"}/></div>
            <div>password:<input type="text" name={"password"}/></div>
            <div>
                <button>Log in</button>
            </div>

        </form>
    </div>
}