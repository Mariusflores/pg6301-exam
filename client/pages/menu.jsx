import {useEffect, useState} from "react";

function useLoading(loadingFunction){
    const [loading, setLoading] = useState(true);
    const [error, setError] = useState();
    const[data, setData] = useState();

    async function load(){

        try{
            setLoading(true);
            setData(await loadingFunction())
        }catch (error){
            setError(error);
        }finally {
            setLoading(false);
        }
    }

    useEffect( () => {
        load();
    }, [])

    return{loading, error, data, reload:load}
}

async function fetchJSON(url){

    const res = await fetch(url)

    if (!res.ok){
        throw new Error(`Failed to load ${res.status} : ${res.statusText}`)
    }

    return await res.json();
}

export function MenuPage(){

    const {loading, error, data} = useLoading(
        async() => fetchJSON("/api/dishes")
    )

    if (loading){
        return <div>Loading..</div>
    }
    if (error){
        return <div>
            <h1>Error</h1>
            <div>{error.toString()}</div>
        </div>

    }


    return <div>

        <h1>Menu</h1>

        <ul>
            {data.map(dish => (
                <li key  ={dish.id}>{dish.dish}</li>
            ))}
        </ul>

        <form >
            <h1>Add new dish</h1>


        </form>
    </div>
}