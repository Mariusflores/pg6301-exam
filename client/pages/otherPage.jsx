import * as React from "react";

export function OtherPage() {
    return <div>
        <h1>Welcome to the other page!</h1>

        <p>Not much to do here eitherðŸ˜‚</p>
    </div>;
}