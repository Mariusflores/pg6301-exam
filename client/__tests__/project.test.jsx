import React from "react";

import {render} from "react-dom";
import {FrontPage} from "../pages/frontPage";
import {OtherPage} from "../pages/otherPage";

describe("project", () => {
    it('shows other page', () => {
        const element = document.createElement("div");
        render(<FrontPage/>, element)

        expect(element.querySelector("h1").innerHTML)
            .toEqual("Welcome to Kristiania Catering!");
        expect(element.innerHTML).toMatchSnapshot();
    });

});




