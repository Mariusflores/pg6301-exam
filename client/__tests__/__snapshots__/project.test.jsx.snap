// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project shows other page 1`] = `"<div><h1>Welcome to Kristiania Catering!</h1></div>"`;
